cniconf/watcher.go

```diff
@@ -78,7 +83,39 @@ func (w *watcher) onChange(version string) error {
 	return nil
 }
 
-func (w *watcher) apply(network metadata.Network) error {
+func (w *watcher) inserthostSubnetRules(host metadata.Host, content []byte) ([]byte, error) {
+	subnet, _ := host.Labels[hostSubnetLabel]
```

Comment:
These settings should only be applied if the networking type is also host-subnet.  For example, if I have vxlan or ipsec networking turned on, you shouldn't inject these settings if the labels are on the host.